src/flag_gems/utils/pointwise_dynamic.py

```diff
@@ -132,627 +168,536 @@ def num_output_tensors(self) -> int:
+                    non_tensor_index += 1
 
-def output_ref_for_wrapper(op_desc: OPDesc) -> str:
-    """Generate output variable refernece for wrapper function.
-    Example: out0, out1
-    """
-    parameters: List[str] = [f"out{i}" for i in range(op_desc.num_outputs())]
-    return ", ".join(parameters)
-
-
-def docstring_for_functional_wrapper(op_desc: OPDesc):
-    doc = f'"""Generated wrapper function with {str(op_desc)}"""'
-    return doc
-
-
-def docstring_for_destination_passing_wrapper(op_desc: OPDesc):
-    doc = f'"""Generated wrapper function with {op_desc.signature(outputs_in_arg=True)}"""'
-    return doc
-
-
-def generate_imports(code: IndentedBuffer) -> IndentedBuffer:
-    code.writeline("import math")
-    code.writeline("import torch")
-    code.writeline("import triton")
-    code.writeline("from triton import language as tl")
-    code.newline()
-    code.writeline("from flag_gems.utils.shape_utils import (")
-    code.writeline("    broadcast_shapes,")
-    code.writeline("    broadcasted_stride,")
-    code.writeline("    c_contiguous_stride,")
-    code.writeline("    volume,")
-    code.writeline("    Stride,")
-    code.writeline(")")
-    code.writeline("from flag_gems.utils.libentry import libentry")
-    code.writeline("from flag_gems.utils.type_utils import type_promotion")
-    code.writeline("import torch._prims_common as utils")
-    code.newline()
-    code.newline()
-    return code
-
-
-def generate_functional_pointwise_wrapper(
-    op_desc: OPDesc,
-    wrapper_name: str,
-    destination_passing_func_name: str,
-    code: IndentedBuffer,
-) -> IndentedBuffer:
-    # wrapper signature
-    parameters: str = parameter_for_wrapper(op_desc, include_outputs=False)
-    wrapper_signature: str = f"def {wrapper_name}({parameters}):"
-    code.writeline(wrapper_signature)
-
-    with code.indent():
-        # docstring
-        wrapper_docstring = docstring_for_functional_wrapper(op_desc)
-        code.writeline(wrapper_docstring)
-
-        shapes_str = ", ".join(
-            f"in{i}.shape" for i in range(op_desc.num_input_tensors())
-        )
-        code.writeline(f"shape = broadcast_shapes([{shapes_str}])")
+            # signature: output ptrs
+            for i in range(schema.num_outputs()):
+                code.writeline(
+                    f"out{output_tensor_index}_ptr: tl.tensor, # of tl.pointer_type"
+                )
+                output_tensor_index += 1
+
+            # signature: strides, for each tensor arguments
+            ndim = self.ndim
+            if ndim > 0:
+                # strides for inputs
+                for i in range(schema.num_input_tensors()):
+                    stride_args = _cs(f"in{i}_stride{j}: int" for j in range(ndim))
+                    code.writeline(f"{stride_args}, # strides for in{i}")
+                    stride_order_args = _cs(
+                        f"in{i}_stride_order{j}: tl.constexpr" for j in range(ndim)
```

Comment:
If `input transpose` is not currently supported, `stride_order` can just be set in load/store blockptr, just 

```
bptr = tl.make_blockptr(inptr, ... order=(rank-1, rank-2, .., 0))
```

Reply:
We now argsort each tensor's strides to get the real stride order. Is this well-supported?

Reply:
Yes. Is there test coverage for scenarios similar to input transpose, such as when the input has shapes: (8, 16) and strides: (1, 8)?

Reply:
There is no such test cases now. I will add some cases with different stride orders.

Reply:
Done